sdk/communication/communication-administration/src/phoneNumber/lro/util.ts

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { PhoneNumberSearch, SearchStatus } from "../..";
+
+export const isComplete = (results: PhoneNumberSearch, status: SearchStatus): boolean | never => {
+  if (results.status !== status) {
+    throw new Error(JSON.stringify(results));
+  }
+
+  return true;
```

Comment:
I don't understand this logic here. Where is the logic that checks that the status is a terminal status?